### ISSUE TITLE
Add stream recording

### ISSUE THREAD

**Issue by ghost** [datarhei/restreamer#692]
**Is your feature request related to the issue? Please describe.**
No

**Describe your desired solution**
Stream recording allows you to record a stream regardless of the source. The recording goes to disk, after which you can take the recording of the stream to your disk, where the user’s stream was originally broadcast

**Describe alternatives you have considered**
No

**Additional context**
No

**Comment by ioppermann** [comment]
Thanks for this suggestion. We added it to our feature list: https://github.com/datarhei/restreamer/discussions/345

**Comment by brixxnatt** [comment]
I hope this comes soon

**Comment by LuizStSantos** [comment]
I liked this feature, and I have ideas for it.

1º - 24-hour recording regardless of whether the RTMP, SRT stream or USB and Blackmagic or HLS inputs drop.

2º  - Location where to save the file.

3º -  Recording start and end times - you can select to do this every day or just specific days.

4º -  Overlay with system time and date, with top left, bottom left, top right and bottom right options.

5º - Maximum time that recordings will be saved, as if it were a DVR with a specific date Example 15 days, 30 days and 90 days.

**Comment by vinattay** [comment]
Great ideas! One additional suggestion: it would be very helpful to allow saving recordings to different storage backends — for example, Amazon S3, Google Cloud Storage, or other external services (including S3-compatible ones like MinIO, Wasabi, Backblaze B2, etc.). In our use case, we need to store recordings off-site for redundancy and long-term storage. This would give more flexibility, especially for cloud-based or distributed setups. Thanks for the awesome work!

**Comment by Magicrafter13** [comment]
Does anyone have suggestions for alternatives until this is implemented?

I'm trying to use the mediamtx project to host a stream receiver because it has built-in recording capabilities, but the files it produces seem to have issues.

**Comment by spprod35** [comment]
I agree with adding the recording functionality to the streaming server. This feature will allow users to save live streams for later viewing, enhancing flexibility and accessibility. It also provides content creators with valuable material for repurposing and analysis, and ensures that important broadcasts are preserved
